# Mission Config of the sub
# ALL UNITS IN M

# priors: [x,y,z, frame], if frame is omitted, the prior will default to leviathan/description/odom
# manager_info docs
# timeout_secs: # of seconds before timing out, if 0 the task won't timeout
# outcome_* : how the manager should respond when a task exits with this outcome
#            skip : move on to the next task
#            later : move this task to the end of the mission_tasks queue, essentially coming back to the task later
#            uh_oh : or any unrecognized string will crash and stop the SM

using_sim_params : True
manager_task_status_pub_rate: 1

startup_delay: 0

tasks:
 startup:
  manager_info:
    timeout_secs: 5
  depth: -1.5
  dive_time: 3

 start_gate:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'uh_oh'
  prior: [-18,-22,-1.5, "world"] # TRANSDEC
  # prior: [4, 0, -1.5, "world"] # TRANSDEC, DIVEWELL 8,0,-1.5  
  dist_behind_gate: 1.0
  third_leg_adjustment: 0.75
  replan_thresh: 1.0
  style_dist: 2.0
  three_leg_depth: -1.0
  do_style: True
  spin_carrot_rads: 0.7

 path0:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  # prior: [-13,25,-4.5, "world"] # Won't matter much b/c we'll almost immediately see it after the SG
  prior: [4.0,0.0,-1.5, "world"] # DIVEWELL
  target_pixel_threshold: 30   # Goal box size around our target pixel for visual servoing, if inside the box we will be successful
  centering_time: 0            # Time between first centered feedback & when we check if we're centered again
  do_orientation: false
  prior_adjustment_dist: 0     # Set to false to not affect the next prior

 jiangshi:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip'
    outcome_timed_out: 'skip'
  # prior: [4.0,0.0,-1.5, "world"] # DIVEWELL
  prior: [-25,-16,-2, "world"] # TRANSDEC
  method: vs        # solvepnp OR vs
  solvepnp:
    use_buoys_yaw: False   # If false, the sub will assume the buoy is facing twd sub
    replan_thresh: 1.0
    slay_dist: 0.0
    approach_dist: 2.0
  visual_servo: # approach and slay
    goal_percentage_of_frame: 0.4    # Percentage of image for target class
    slay_timeout: 10                 # How long the slay will last
    slay_carrot: 0.7                 # controls the speed when slaying the buoy by adjusting drive setpoint
    target_pixel_threshold: 50    # how close to our target do we need to be    

 triangle:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  prior: [-29,-18, -2.5, "world"]
  do_orbit: false
  orbit_right: true
  strafe_carrot: 0.2
  target_vampire: 'vampire_fathead'
  visual_servo: # approach and slay
    goal_percentage_of_frame: 0.3    # Percentage of image for target class
    slay_timeout: 10                 # How long the slay will last
    slay_carrot: 0.7                 # controls the speed when slaying the buoy by adjusting drive setpoint
    target_pixel_threshold: 50    # how close to our target do we need to be    
  mikes_method:
    do: false
    go_over_jiangshi_depth: -0.7
    meters_fwd: 2.0

 torpedos:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  prior: [-2.5,2.0,-1.5, "world"] # Won't matter much b/c we'll almost immediately see it after the SG
  target_pixel_threshold: 20   # Goal box size around our target pixel for visual servoing, if inside the box we will be successful

 dropper:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  prior: [-33, -11, -2.0, "world"] # TRANSDEC
  # prior: [4, 0.0, -1, "world"] # DIVEWELL
  target_pixel_threshold: 30   # Goal box size around our target pixel for visual servoing, if inside the box we will be successful
  approach_depth: -1
  drop_timeout: 45
  drive_carrot: 2.0
  depth_carrot: 0.5
  drop_depth: -2.5
  drop_depth_timeout: 60

 debug:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  centering_time: 5            # Time between first centered feedback & when we check if we're centered again
  target_pixel_threshold: 30   # Goal box size around our target pixel for visual servoing, if inside the box we will be successful
  drop_depth: -2.5

 octagon:
  manager_info:
    timeout_secs: 0
    outcome_not_found: 'skip' # skip, later
    outcome_timed_out: 'skip'
  prior: [-38, 0.0, -1, "world"] 
  target_pixel_threshold: 20   # Goal box size around our target pixel for visual servoing, if inside the box we will be successful
  approach_depth: -1
  depth_carrot: 0.2
  drive_carrot: 1.0
  octagon_height: 0.0
  finish_depth: -1.5
  finish_depth_thresh: 0.4

mission_tasks: # Tasks To Do In Order
 - triangle
#  - startup
#  - start_gate # maybe do visual servoing on the gate?
#  - octagon
#  - dropper

#  - jiangshi
